### PR TITLE
Changes 1559 to have a single shared pool for all transactions.

### PR DESCRIPTION
**Pull request by MicahZoltu**
Rather than having two pools and a migration, we instead have a single transaction pool and legacy transactions are "upgraded" into 1559 transactions using a *very* naive strategy of just setting `fee_cap == gas_bribe == gas_price`.

While in here, I also made a few other minor changes since this is a consensus change and will require all clients update:
1. Renamed premium to bribe.  I think this makes it more clear what this is, it is a bribe for the miner to convince them to include your transaction.
2. Changed the gas target calculation to be a bit simpler (per prior conversation with @i-norden)
3. Added special handling for when the gas_used == gas_target (no change in target for next block).
4. Adds missing assertion that gas_used wasn't more than 2x the gas target.
5. Added clarity to some names like `fee_cap` to `fee_cap_per_gas` and `miner_bribe` to `miner_bribe_per_gas`.

### CODE REVIEW

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 1559 requires approval from one of (@i-norden, @mslipper, @afdudley, @vbuterin, @econoar)

File: EIPS/eip-1559.md
Comment (bluepintail) [inline EIPS/eip-1559.md:19]: `FEECAP` was previously the absolute maximum that the transaction sender is willing to pay. I think setting the overall maximum and the miner bribe maximum (as previously specified) makes more sense, otherwise the user has to add the two together to work out what the overall maximum fee is.

Comment (MicahZoltu) [reply EIPS/eip-1559.md:19]: There was no intention to change the behavior here (this particular change was meant to be non-normative).  Do you have a suggestion for better wording than what I have that would make that more clear (ideally without going *too* in depth)?

Comment (bluepintail) [reply EIPS/eip-1559.md:19]: Might be easier to specify the other way around - "transactions specify the maximum fee per gas they are willing to give to miners to incentivize them to include their transaction (aka: miner bribe), they also specify a maximum overall fee per gas they are willing to pay, including both the miner bribe and the network fee (aka: base fee)"

Comment (MicahZoltu) [reply EIPS/eip-1559.md:19]:
```suggestion
Transactions specify the maximum fee per gas they are willing to give to miners to incentivize them to include their transaction (aka: miner bribe).
Transactions also specify the maximum fee per gas they are willing to pay total (aka: fee cap), which covers both the miner bribe and the block's network fee per gas (aka: base fee).
```
How does this sound @bluepintail

**Comment by i-norden** [comment]
I think instead of `fee_cap == gas_bribe == gas_price`  

It is more accurate to say `fee_cap == gas_price; gas_bribe = fee_cap - base_fee`

In practice we should set `gas_bribe = gas_price` in the tx struct, but the amount of `gas_price` actually available for the bribe after burning the` base_fee` will be `fee_cap - base_fee`.

Thanks for opening this! I haven't had a chance to look into the rest of this PR in more depth, will do so soon.

**Comment by MicahZoltu** [comment]
@i-norden I think what you are getting at is that `miner_bribe_per_gas` is a bit of a misnomer and really it is `maximum_miner_bribe_per_gas`.  If so, I agree with you and have updated the EIP to reflect that.

Now that we have merged the other PR, I have rebased this and it is ready to merge once the authors approve it.  Per discussion in Discord, we should make note that `711474afe36a9d778f7e1858b2c9a88b0bf146d0` is the commit hash for EIP-1559 that is currently on the integration testnet, and whatever the merge commit hash for this is will be the "target" EIP we are working toward and probably what new clients should implement.

@i-norden, @mslipper, @AFDudley, @vbuterin, @econoar

File: EIPS/eip-1559.md
Comment (vbuterin) [inline EIPS/eip-1559.md:100]:
```suggestion
ELASTICITY_MULTIPLIER = 2
```

File: EIPS/eip-1559.md
Comment (vbuterin) [inline EIPS/eip-1559.md:110]:
```suggestion
		assert block.gas_used <= block.gas_target * ELASTICITY_MULTIPLIER, 'invalid block: too much gas used'
```

**Comment by MicahZoltu** [comment]
I have removed the prose text that referred to the transition as well (missed it in original PR).

File: EIPS/eip-1559.md
Comment (tkstanczak) [inline EIPS/eip-1559.md:123]: need to address cases when expected_base_fee ends up being < 0

File: EIPS/eip-1559.md
Comment (tkstanczak) [inline EIPS/eip-1559.md:118]: max not needed here, it is fine to move by near 0 wei gas prices

File: EIPS/eip-1559.md
Comment (MicahZoltu) [reply EIPS/eip-1559.md:123]: Discussed this in Discord, recording result here for thoroughness: In that particular conditional branch, there is no way for `gas_delta` to be negative, thus there is no way for `fee_delta` to be negative.  The same is true for the other branch.

**Review by abdelhamidbakhta — APPROVED** [review]
Looks good to me.

**Pull request merged**
